pkg/rpc/request/param.go

```diff
@@ -191,7 +191,11 @@ func (p *Param) UnmarshalJSON(data []byte) error {
 			case *NotificationFilter:
 				p.Value = *val
 			case *ExecutionFilter:
-				p.Value = *val
+				if (*val).State == "HALT" || (*val).State == "FAULT" {
```

Comment:
I think there is no need in explicit pointer dereference. And it can be more compact, if condition is inverted, `else` can be dropped.

Reply:
Well, kinda, but probably not worth it.